spring-cloud-dataflow-server-core/src/test/java/org/springframework/cloud/dataflow/server/service/impl/DefaultSchedulerServiceTests.java

```diff
@@ -431,18 +436,51 @@ public void testScheduleWithoutCommandLineArguments() {
 	@Test
 	public void testGetDefaultCTR() {
 		ScheduleRequest request = getScheduleRequest(new ArrayList<>(), "springcloudtask/composed-task-runner:latest", "1: timestamp && 2: timestamp");
-		AppDefinition definition = request.getDefinition();
 		assertEquals("Docker Resource [docker:springcloudtask/composed-task-runner:latest]", request.getResource().toString());
 	}
+	@Test
+	public void testVersionWithResource() {
+		String validVersionNumber = "3.0.0";
+		ScheduleRequest request = scheduleRequest(validVersionNumber);
+		assertThat(request.getResource().toString()).contains("file:src/test/resources/apps/foo-task");
+	}
+
+	@Test
+	public void testVersionWithResourceInvalidVersion() {
+		String invalidVersionNumber = "2.0.0";
+		assertThatExceptionOfType(IllegalArgumentException.class)
+			.isThrownBy(() -> {
+				scheduleRequest(invalidVersionNumber);
+			}).hasToString("java.lang.IllegalArgumentException: Unknown task app: demo");
+	}
+
+	private ScheduleRequest scheduleRequest(String appVersionToTest) {
+		String definition = "demo";
+		Map<String, String> resourceTestProps = new HashMap<>(testProperties);
```

Comment:
I am enjoying the newer `Map.of(` API. Combined w/ `var` it makes the code a bit less terse.